cmd/nodecmd/create.go

```diff
@@ -0,0 +1,377 @@
+		certFilePath, err := app.GetSSHCertFilePath(certName)
+		if err != nil {
+			return err
+		}
+		if app.CheckCertInSSHDir(certFilePath) {
+			ux.Logger.PrintToUser(fmt.Sprintf("Using existing key pair %s on AWS", keyPairName))
+			useExistingKeyPair = true
+		} else {
+			ux.Logger.PrintToUser(fmt.Sprintf("Default Key Pair named %s already exists", keyPairName))
+			ux.Logger.PrintToUser(fmt.Sprintf("We need to create a new Key Pair on AWS as we can't find Key Pair named %s on AWS", keyPairName))
+			keyPairName, err = getNewKeyPairName(ec2Svc)
+			if err != nil {
+				return err
+			}
+			certName = keyPairName + "-kp.pem"
+			terraform.SetKeyPair(rootBody, keyPairName, certName)
+		}
+	}
+	securityGroupExists, sg, err := checkSecurityGroupExists(ec2Svc, securityGroupName)
+	if err != nil {
+		return err
+	}
+	userIPAddress, err := getIPAddress()
+	if err != nil {
+		return err
+	}
+	if !securityGroupExists {
+		ux.Logger.PrintToUser(fmt.Sprintf("Creating new security group %s on AWS", securityGroupName))
+		terraform.SetSecurityGroup(rootBody, userIPAddress, securityGroupName)
+	} else {
+		ux.Logger.PrintToUser(fmt.Sprintf("Using existing security group %s on AWS", securityGroupName))
+		ipInTCP, ipInHTTP := checkCurrentIPInSg(sg, userIPAddress)
+		terraform.SetSecurityGroupRule(rootBody, userIPAddress, *sg.GroupId, ipInTCP, ipInHTTP)
+	}
+	terraform.SetElasticIP(rootBody)
+	terraform.SetUpInstance(rootBody, securityGroupName, useExistingKeyPair, keyPairName, ami)
+	terraform.SetOutput(rootBody)
+	err = terraform.SaveTerraformFile(tfFile, hclFile)
+	if err != nil {
+		return err
+	}
+	instanceID, elasticIP, err := terraform.RunTerraform()
+	if err != nil {
+		return err
+	}
+	ux.Logger.PrintToUser("A new EC2 instance is successfully created on AWS!")
+	certFilePath, err := app.GetSSHCertFilePath(certName)
+	if err != nil {
+		return err
+	}
+
+	if !useExistingKeyPair {
+		err = addCertToSSH(certName)
+		if err != nil {
+			return err
+		}
+	}
+
+	inventoryPath := app.GetAnsibleInventoryPath(clusterName)
+	if err := ansible.CreateAnsibleHostInventory(inventoryPath, elasticIP, certFilePath); err != nil {
+		return err
+	}
+	time.Sleep(5 * time.Second)
+
+	ux.Logger.PrintToUser("Installing AvalancheGo and Avalanche-CLI and starting bootstrap process on the newly created EC2 instance...")
+	if err := ansible.RunAnsibleSetUpNodePlaybook(inventoryPath); err != nil {
+		return err
+	}
+	err = createNodeConfig(instanceID, region, ami, keyPairName, certFilePath, securityGroupName, elasticIP, clusterName)
+	if err != nil {
+		return err
+	}
+	PrintResults(instanceID, elasticIP, certFilePath, region)
+	ux.Logger.PrintToUser("AvalancheGo and Avalanche-CLI installed and node is bootstrapping!")
+	err = terraform.RemoveExistingTerraformFiles()
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func requestAWSAccountAuth() error {
+	confirm := "Do you authorize Avalanche-CLI to access your AWS account to set-up your Avalanche Validator node?"
```

Comment:
Should probably add a bit more scare language to say that this will result in charges to their account and Ava Labs is not liable for any bills incurred.

Reply:
We probably want to run by legal eventually, so not a super big deal atm